tests/core/test_utils.py

```diff
@@ -46,3 +46,21 @@ def test_mesh_repair(getAssetPath):
 
     with pytest.raises(ValueError):
         repairMesh(plane)
+
+
+@pytest.mark.slow
+def test_unify_mesh():
+    # Create nested sphere
+    nested_sphere = (
+        trimesh.creation.icosphere(radius=5)
+        .difference(trimesh.creation.icosphere(radius=4))
+        .union(trimesh.creation.icosphere(radius=3))
+        .difference(trimesh.creation.icosphere(radius=2))
+    )
+
+    # Manually append a box
+    bad_mesh = trimesh.util.concatenate(
+        nested_sphere, trimesh.creation.box(bounds=((0, 0, 0), (3, 5, 3)))
+    )
+
+    unifyMesh(bad_mesh)
```

Comment:
Would be nice to add some assertions here, e.g. that the unified mesh is a valid volume and has only one body.

Reply:
Good idea on the volume check, but unfortunately the unified mesh can actually still have two bodies and be valid (think two discrete spheres).

Reply:
Good idea, though this one actually ends up with 3 bodies so I'll make the assertion reflect that.

Reply:
Hmm, this is actually something we should discuss since this goes from 5 "bodies" (by face adjacency) to 3.